public/app/features/dashboard/custom_time_ranges/TimeRangeEditor.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { PureComponent } from 'react';
+
+export interface Props {
+  range: any;
+  mode: string;
+  add: any;
+  update: any;
+}
+
+export default class TimeRangeEditor extends PureComponent<Props> {
+  constructor(props) {
+    super(props);
+
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    // Set the state
+    console.log(this.props.range);
+    this.state = { mode: this.props.mode };
+    if (this.props.mode === 'edit') {
+      console.log('Edit Mode', this.props.range);
+    } else {
+      console.log('Not Edit Mode', this.props.range);
+    }
+  }
+
+  add = () => {
+    this.props.add(this.state);
+  };
+  update = () => {
+    this.props.update(this.state);
+  };
+  onChange(e) {
```

Comment:
Please use arrow functions for these functions. And remove the constructor after that.